File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        long num = 0;
        long sum = 0;

        while(n>0){
            long rem = n%10;

            if(rem!=0){
                sum+=rem;
                num = num*10 + rem ; 
            }
            n = n/10;
        }
        long ans = 0;

        while(num>0){
            long rem2 = num%10;
            ans = ans*10 + rem2;
            num/=10;
        }

        return ans*sum ;
    }
}

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        n = str(n)
        curr_sum = 0
        s = ""

        if int(n) == 0:
            return 0
        
        for i in range(0, len(n)):
            if n[i] == "0":
                continue
            
            curr_sum += int(n[i])
            s += n[i]

        return int(s) * curr_sum